src/interfaces/coral_web/src/pages/agents/[[...slug]].tsx

```diff
@@ -183,12 +181,29 @@ export const getServerSideProps: GetServerSideProps = async (context) => {
     cohereClient: CohereClient;
   };
 
-  const conversationId = context.params?.conversationId as string;
+  const { conversationId, agentId } = getSlugRoutes(context.query.slug);
+
+  if (!conversationId && !agentId && context.resolvedUrl !== '/agents') {
+    return {
+      redirect: {
+        destination: '/agents',
+        permanent: false,
+      },
+    };
+  }
 
   await Promise.allSettled([
+    deps.queryClient.prefetchQuery({
+      queryKey: ['agent', agentId],
+      queryFn: async () => {
+        if (!agentId) return;
```

Comment:
nit: maybe we can use `enabled` instead of checking if the agentId is defined.

Reply:
not possible with prefetchQuery =/